src/freenet/node/DarknetPeerNode.java

```diff
@@ -1393,12 +1395,23 @@ public int sendDownloadFeed(FreenetURI URI, String description) {
 
 	public int sendTextFeed(String message) {
 		long now = System.currentTimeMillis();
-		SimpleFieldSet fs = new SimpleFieldSet(true);
-		fs.put("type", Node.N2N_TEXT_MESSAGE_TYPE_USERALERT);
-		fs.putSingle("text", Base64.encodeUTF8(message));
-		fs.put("composedTime", now);
-		sendNodeToNodeMessage(fs, Node.N2N_MESSAGE_TYPE_FPROXY, true, now, true);
-		this.setPeerNodeStatus(System.currentTimeMillis());
+		long msgid = node.random.nextLong();
+		// split large messages
+		int requiredN2nCount = 1 + (message.length() / 1024);
+		String messagePart;
+		for (int i = 0; i < requiredN2nCount; i++) {
+			messagePart = message.substring(i * 1024,
+							Math.min((i+1) * 1024,
```

Comment:
Non-blocker:
You can use "import static" to be able to use self-explanatory functions like min() without prefixing them with the parent class' name. Java isn't that noisy anymore when using all its features to full extent....
In Eclipse putting the cursor on the function name and pressing CTRL+SHIFT+M automatically adds the import.